common/kafka/kafka_consumer.h

```diff
@@ -72,36 +70,33 @@ class KafkaConsumer {
   // Return set of topics as a string for logging
   const std::string& GetTopicsString() const;
 
-  virtual ~KafkaConsumer() {
-    if (consumer_) {
-      consumer_->close();
-    }
-  }
+  virtual ~KafkaConsumer();
 
   const std::string partition_ids_str_;
 
 private:
-  // For mock ONLY
-  explicit KafkaConsumer(const std::unordered_set<std::string>& topic_names)
-      : partition_ids_str_("0"),
-        topic_names_(topic_names),
-        partition_ids_({0}),
-        kafka_consumer_type_metric_tag_("") {}
+  KafkaConsumer(RdKafkaConsumerHolder* holder,
+                const std::unordered_set<uint32_t>& partition_ids,
+                const std::unordered_set<std::string>& topic_names,
+                const std::string& kafka_consumer_type);
 
   virtual bool SeekInternal(const std::unordered_set<std::string>& topic_names,
                             const int64_t timestamp_ms);
 
   virtual bool SeekInternal(const std::map<std::string, std::map<int32_t,
-      int64_t>>& last_offsets);
+    int64_t>>& last_offsets);
 
-  std::unordered_set<std::string> topic_names_;
+  const std::unordered_set<std::string> topic_names_;
   std::string topic_names_string_;
   const std::unordered_set<uint32_t> partition_ids_;
-  std::shared_ptr<RdKafka::KafkaConsumer> consumer_;
+  const std::shared_ptr<RdKafkaConsumerHolder> consumer_;
   // Tag used when logging metrics, so we can differentiate between kafka
   // consumers for different use cases.
   const std::string kafka_consumer_type_metric_tag_;
+  std::shared_ptr<common::MultiFilePoller::CallbackId> cbIdPtr_;
```

Comment:
Can we use the snake_case naming style to keep it consistent? 
Also, how about renaming to `callback_id_`?

Reply:
Sure